department-of-veterans-affairs/vets-website
# Burials | Date alert message isn't announced to screen reader users

## Original Issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/42249

[URL of fix](/burials-and-memorials/application/530/veteran-information/burial)

## Expected behavior
When filling out the Veterans Burial Date, if the user enters a date that is more than 2 years ago, a `va-alert` will appear below the date fields informing them that the burial date must be within 2 years of todays date in order for them to be eligible to receive a non-service-connected allowance. The alert content should be read aloud by a screenreader

## Current behavior
Currently when the user enters a date that is more than 2 years ago, the alert appears, but the screenreader ignores it

## Your fix
I updated the code to use `va-alert` instead of the `usa-alert` CSS class and added `aria-live="polite"` to it. I also had to delay the rendering of the alert by 1000ms because VoiceOver (on macOS at least) reads the last number entered into the year input, followed by the whole number. Without the delay, VoiceOver will read the last number, then start to read the alert content, then immediately read the whole year input.

I also made some updates to the styling as part of the tickets request as well

## How has this been tested?
* Navigate to `/burials-and-memorials/application/530/veteran-information/burial`
* Activate VoiceOver
* Enter a date more than 2 years prior to todays date in the Date of Burial field
* VoiceOver should read the last number entered, then the whole year, then the alert content

## Screenshots
### Current behavior
![current](https://user-images.githubusercontent.com/13838621/176015548-f13eea46-ac9d-460a-9ad5-5dc82cf9ed71.gif)

### Expected behavior
![delayed](https://user-images.githubusercontent.com/13838621/176013785-769f99a1-0e95-40e6-bb13-664286d4bd97.gif)

## Acceptance criteria
- [x] The dynamic alert is announced using `aria-live="polite"`
- [x] An icon with alt text or a text label is provided to represent the urgency the background color yellow communicates (e.g. "**Notice**: If filing for a non-service-connected allowance, the Veteran’s burial date must be no more than 2 years from the current date. Find out if you still qualify..."
- [x] The link text should include "(opens in new tab)" to give magnification users a heads up that it will be opening in a new tab


## Definition of done
